java/src/main/java/cucumber/runtime/java/ObjectFactoryLoader.java

```diff
@@ -53,6 +55,46 @@ public static ObjectFactory loadObjectFactory(ClassFinder classFinder, String ob
         return objectFactory;
     }
 
+    private static io.cucumber.core.backend.ObjectFactory loadSingleObjectFactory(final Reflections reflections, final ServiceLoader<io.cucumber.core.backend.ObjectFactory> loader) {
+        final Map<String, io.cucumber.core.backend.ObjectFactory> objectFactories = loadObjectFactories(reflections, loader);
+        final io.cucumber.core.backend.ObjectFactory result;
+        if (objectFactories.isEmpty()) {
+            throw new NoInstancesException(io.cucumber.core.backend.ObjectFactory.class);
+        } else if (objectFactories.size() > 1) {
+            throw new TooManyInstancesException(objectFactories.values());
+        } else {
+            result = objectFactories.values().iterator().next();
+        }
+        return result;
```

Comment:
It's okay to use an early return statement once an object factory has been found.